pkg/controller/bundle/bundle_unpacker.go

```diff
@@ -768,3 +769,31 @@ func getCondition(job *batchv1.Job, conditionType batchv1.JobConditionType) (con
 	}
 	return
 }
+
+// OperatorGroupBundleUnpackTimeout returns bundle timeout from annotation if specified.
+// This is to overrides the --bundle-unpack-timeout flag value on per-OperatorGroup basis.
+// If the timeout cannot be parsed it's set to < 0 and subsequently ignored.
+func OperatorGroupBundleUnpackTimeout(logger *logrus.Logger, ogLister v1listers.OperatorGroupNamespaceLister) (time.Duration, error) {
+	ignoreTimeout := -1 * time.Minute
+
+	ogs, err := ogLister.List(k8slabels.Everything())
+	if err != nil || len(ogs) == 0 {
+		return ignoreTimeout, nil
```

Comment:
Shouldn't this return an `err`?

Reply:
Also, the check for `len(ogs) == 0` matches the check below. Are we treating 0 different than not 1?

Reply:
I wonder if we ought to return the error if the `List` call fails and remove the `|| len(ogs) == 0`. This can let the caller decide whether they want to retry, or are happy with using the default value. Maybe we could also add a ticket for the old code so we get this rechecked. I'm worried silent errors around this code could lead to hard to determine bugs. wdyt?

Reply:
I made this function to return an error from `List` and removed `|| len(ogs) == 0`. From what I know about OLM so far - `OperatorGroup` is required and without it resolution will fail eventually. In case of these errors - sync will be requeued (no extra code on caller required).

I'll try to the same for old code in `IsFailForwardEnabled`, but in a separate PR. Will see how it goes.

Reply:
Created https://github.com/operator-framework/operator-lifecycle-manager/pull/2957 for `IsFailForwardEnabled`.